src/main/java/at/hannibal2/skyhanni/data/PetAPI.kt

```diff
@@ -35,31 +178,492 @@ object PetAPI {
         "➡",
     )
 
-    fun isPetMenu(inventoryTitle: String): Boolean = petMenuPattern.matches(inventoryTitle)
+    fun isPetMenu(inventoryTitle: String, inventoryItems: Map<Int, ItemStack>): Boolean {
+        if (!petMenuPattern.matches(inventoryTitle)) return false
+
+        val goBackLore = inventoryItems[48]?.getLore().orEmpty()
+        return !goBackLore.any { forgeBackMenuPattern.matches(it) }
+    }
 
-    // Contains color code + name and for older SkyHanni users maybe also the pet level
     var currentPet: String?
         get() = ProfileStorageData.profileSpecific?.currentPet?.takeIf { it.isNotEmpty() }
         set(value) {
             ProfileStorageData.profileSpecific?.currentPet = value
         }
 
-    fun isCurrentPet(petName: String): Boolean = currentPet?.contains(petName) ?: false
+    fun isCurrentPet(petName: String): Boolean = pet?.cleanName?.contains(petName) ?: false
 
     fun getCleanName(nameWithLevel: String): String? {
-        petItemName.matchMatcher(nameWithLevel) {
+        petItemNamePattern.matchMatcher(nameWithLevel) {
             return group("name")
         }
-        neuRepoPetItemName.matchMatcher(nameWithLevel) {
+        neuRepoPetItemNamePattern.matchMatcher(nameWithLevel) {
             return group("name")
         }
 
         return null
     }
 
-    fun getPetLevel(nameWithLevel: String): Int? = petItemName.matchMatcher(nameWithLevel) {
+    @Deprecated(message = "use PetAPI.pet.level")
+    fun getPetLevel(nameWithLevel: String): Int? = petItemNamePattern.matchMatcher(nameWithLevel) {
         group("level").toInt()
     }
 
-    fun hasPetName(name: String): Boolean = petItemName.matches(name) && !ignoredPetStrings.any { name.contains(it) }
+    @Deprecated(message = "use PetAPI.pet.name")
```

Comment:
what is the replacement for this function? ignord pet strings is not  getting used by any other function and the petapi pet name call is also not matching this functions purpose (checking if the parameter is a valid pet name)